### DISCUSSION TITLE
How to get historical maximum values?

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#26236] [Q&A General]
For some reason I have to get historical maximum values of elements or nodes and then use them to compute some free energy. But after reading documentation, I find there is seemingly no material object or kernels that can do that. Am I right? And how can I get such maximum values?

**Comment by GiudGiud** [comment]
Hello

Use a MaterialRealAux to store the material property in an auxiliary variable.
Then use a parsed expression with a max() in it in a ParsedAux to keep the max of the current and previous max of the variable.
This will give you a spatially dependent max.
If you need a single value, use an ElementExtremeValue postprocessor on the variable

Guillaume

**Reply by RWTHLHK** [reply]
Thanks! It works.
